modules/json/src/parsers/convert-json-expression.js

```diff
@@ -0,0 +1,74 @@
+import {get} from '../utils/get';
+
+// expression-eval: Small jsep based expression parser that supports array and object indexing
+import expressionEval from 'expression-eval';
+
+class EvaluationError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'EvaluationError';
+  }
+}
+
+const SINGLE_IDENTIFIER_REGEX = /^\s*[A-Za-z]+\s*$/;
+
+const cachedExpressionMap = {
+  '-': object => object
+};
+
+// Calculates an accessor function from a JSON string
+// '-' : x => x
+// 'a.b.c': x => x.a.b.c
+export default function convertJSONExpression(propValue, configuration, isAccessor) {
+  // NOTE: Can be null which represents invalid function. Return null so that prop can be ommitted
+  if (propValue in cachedExpressionMap) {
+    return cachedExpressionMap[propValue];
+  }
+
+  let func;
+  // TODO - backwards compatibility for v6 JSON API: (simple identifier evaluates to row[string])
+  // TODO - Check may not be needed - syntax could overlap with expression-eval?
+  // TODO what should be done in the case of an undefined identifier?
+  const isSingleIdentifier =
+    isAccessor &&
+    SINGLE_IDENTIFIER_REGEX.test(propValue) &&
```

Comment:
I wonder if we could drop the REGEXP and handle this by just inspecting the ast returned by expression-eval. Maybe check if the return AST consists of a single node of type `Identifier`:

https://github.com/donmccurdy/expression-eval/blob/master/index.js#L93